### ISSUE TITLE
Performance drop from version 1.4 onwards

### ISSUE THREAD

**Issue by Jason-Khan** [Lightning-AI/lightning#12925]
## 🐛 Bug

I ran my classification model on the same code, using 3 different versions of PyTorch Lightning.
Red curve is 1.4.8, orange is 1.5.10, and blue is 1.6.2.
Called pl.seed_everything(0) and can confirm that results can be reproduced if run with the same seed again.
![image](https://user-images.githubusercontent.com/17810447/165881872-2f5bfa35-ab56-4624-a704-cb11296c7c0c.png)

Nothing else is changed. 
PyTorch version LTS 1.8.2, batch size 64, gradient accumulation 32.

Did something break after 1.4.8?

### To Reproduce
Sorry unable to provide the private code base.

### Expected behavior
Models should be achieving similar validation performance, but an older version of pytorch lightning achieves 2% higher performance

### Environment

- PyTorch Lightning Version (e.g., 1.5.0): 1.4.8, 1.5.10, 1.6.2
- PyTorch Version (e.g., 1.10): 1.8.2
- Python version (e.g., 3.9): 3.8
- OS (e.g., Linux): Linux
- CUDA/cuDNN version: 11.1
- GPU models and configuration: 1 A100
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @awaelchli

**Comment by carmocca** [comment]
Hi! 1.4.8 was released quite a while ago, so I cannot tell you what could be the problem off the top of my head.

Since you cannot provide a reproducible script, you can try debug the issue by comparing the model outputs for each sample and how many of them you have.

But if validation is not very expensive, the best way would be to git-blame the [list of commits](https://github.com/PyTorchLightning/pytorch-lightning/compare/1.4.8...1.5.0) between both versions until you see which one causes the divergence
